shelley-ma/impl/src/Cardano/Ledger/Allegra/Translation.hs

```diff
@@ -56,9 +74,43 @@ instance Crypto c => TranslateEra (AllegraEra c) NewEpochState where
           nesPd = nesPd nes
         }
 
-instance Crypto c => TranslateEra (AllegraEra c) Tx where
+instance forall c. Crypto c => TranslateEra (AllegraEra c) Tx where
   type TranslationError (AllegraEra c) Tx = ()
-  translateEra _ = error "TODO Shelley to Allegra translation"
+  translateEra ctx (Tx body witness md) =
```

Comment:
Are we guaranteed that the `TxId` are the same?

Reply:
That is the intention of https://github.com/input-output-hk/cardano-ledger-specs/pull/1954 

obviously we'll need good tests!

Reply:
For these eras, yes. The `TxId` type does not change, and the hash (which is in any case just a phantom) is fixed now.